### ISSUE TITLE
how can use syslog as driver for logging?

### ISSUE THREAD

**Issue by ninjatux** [tongueroo/ufo#30]
I would like to use `syslog` as logging driver for my containers, how can I configure that in the task definition and template?

**Comment by tongueroo** [comment]
@ninjatux Believe this is how it is done. Open up `.ufo/templates/main.json.erb` - this gets created as part of the [ufo init](http://ufoships.com/reference/ufo-init/) command.  Then you can adjust the ecs template definition to something like this:

```json
            "logConfiguration": {
                "logDriver": "syslog"
            },
```

It looks like for syslog there might be no options: https://docs.docker.com/config/containers/logging/configure/

Here's the specific aws docs section [Specifying a Log Configuration in your Task Definition](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/using_awslogs.html#specify-log-config)

Also, you might have to enable the log driver by added it the ECS_AVAILABLE_LOGGING_DRIVERS variable to your `/etc/ecs/ecs.config`. Relevant docs:

* https://docs.aws.amazon.com/AmazonECS/latest/developerguide/using_awslogs.html#enable_awslogs
* https://docs.aws.amazon.com/AmazonECS/latest/developerguide/ecs-agent-install.html

Hope that helps.

**Comment by tongueroo** [comment]
@ninjatux This might also help http://ufoships.com/docs/tutorial-ufo-tasks-build/ It covers how task definitions with ufo works. Closed the issue for tidiness. Feel free to reopen if required.

**Comment by ninjatux** [comment]
hi @tongueroo I've already tried to do that, but `ufo` throws some errors when compiling the task.

Here's the output

```
Building Task Definitions...
Generating Task Definitions:
  .ufo/output/service.json
Task Definitions built in .ufo/output
Traceback (most recent call last):
	18: from /opt/circleci/.rvm/gems/ruby-2.5.0/bin/ruby_executable_hooks:15:in `<main>'
	17: from /opt/circleci/.rvm/gems/ruby-2.5.0/bin/ruby_executable_hooks:15:in `eval'
	16: from /opt/circleci/.rvm/gems/ruby-2.5.0/bin/ufo:23:in `<main>'
	15: from /opt/circleci/.rvm/gems/ruby-2.5.0/bin/ufo:23:in `load'
	14: from /opt/circleci/.rvm/gems/ruby-2.5.0/gems/ufo-3.3.1/exe/ufo:14:in `<top (required)>'
	13: from /opt/circleci/.rvm/gems/ruby-2.5.0/gems/thor-0.20.0/lib/thor/base.rb:466:in `start'
	12: from /opt/circleci/.rvm/gems/ruby-2.5.0/gems/ufo-3.3.1/lib/ufo/command.rb:43:in `dispatch'
	11: from /opt/circleci/.rvm/gems/ruby-2.5.0/gems/thor-0.20.0/lib/thor.rb:387:in `dispatch'
	10: from /opt/circleci/.rvm/gems/ruby-2.5.0/gems/thor-0.20.0/lib/thor/invocation.rb:126:in `invoke_command'
	 9: from /opt/circleci/.rvm/gems/ruby-2.5.0/gems/thor-0.20.0/lib/thor/command.rb:27:in `run'
	 8: from /opt/circleci/.rvm/gems/ruby-2.5.0/gems/ufo-3.3.1/lib/ufo/cli.rb:53:in `ship'
	 7: from /opt/circleci/.rvm/gems/ruby-2.5.0/gems/ufo-3.3.1/lib/ufo/tasks/builder.rb:9:in `ship'
	 6: from /opt/circleci/.rvm/gems/ruby-2.5.0/gems/ufo-3.3.1/lib/ufo/tasks/register.rb:9:in `register'
	 5: from /opt/circleci/.rvm/gems/ruby-2.5.0/gems/ufo-3.3.1/lib/ufo/tasks/register.rb:9:in `each'
	 4: from /opt/circleci/.rvm/gems/ruby-2.5.0/gems/ufo-3.3.1/lib/ufo/tasks/register.rb:12:in `block in register'
	 3: from /opt/circleci/.rvm/gems/ruby-2.5.0/gems/ufo-3.3.1/lib/ufo/tasks/register.rb:26:in `register'
	 2: from /opt/circleci/.rvm/gems/ruby-2.5.0/gems/ufo-3.3.1/lib/ufo/tasks/register.rb:40:in `fix_log_configuation_option'
	 1: from /opt/circleci/.rvm/gems/ruby-2.5.0/gems/ufo-3.3.1/lib/ufo/tasks/register.rb:40:in `each'
/opt/circleci/.rvm/gems/ruby-2.5.0/gems/ufo-3.3.1/lib/ufo/tasks/register.rb:43:in `block in fix_log_configuation_option': undefined method `delete' for nil:NilClass (NoMethodError)
```

and this is the task template I am using

```json
{
  "family": "<%= @family %>",
  "containerDefinitions": [
    {
      "name": "<%= @name %>",
      "image": "<%= @image %>",
      "cpu": <%= @cpu %>,
      <% if @memory %>
      "memory": <%= @memory %>,
      <% end %>
      <% if @memory_reservation %>
      "memoryReservation": <%= @memory_reservation %>,
      <% end %>
      "command": <%= @command.to_json %>,
      <% if @environment %>
      "environment": <%= @environment.to_json %>,
      <% end %>
      "logConfiguration": {
        "logDriver": "syslog"
      },
      "essential": true
    }
  ]
}
```

Aas you see it's pretty basic. Can you help me understand why this fails? It seems it's missing something from the template, but I am not sure about that.

**Comment by ninjatux** [comment]
if I comment the line `26` in `register.rb` (`data = fix_log_configuation_option(data)`) it works :)

I'll investigate further what is breaking it, but ruby is not my thing :P

**Comment by ninjatux** [comment]
forgot to mention that I am using version `3.3.1`, anyway problem persists also in `3.4.0`

**Comment by tongueroo** [comment]
@ninjatux I see! Your configuration is correct. It's a ufo bug. Sorry! Thanks for pointing out. Yup, that method dasherizes the log configuration portion of the task definition json; AWS uses different casing for that portion. Looking at it now.

**Comment by tongueroo** [comment]
@ninjatux Fixed with PR #32 and released in version 3.4.1 https://github.com/tongueroo/ufo/blob/master/CHANGELOG.md 

Feel free to re-open if needed.

**Comment by tongueroo** [comment]
Improvements to log configuration handling also made in #32
